File: Two_Pointers/088.Merge-Sorted-Array/088.Merge-Sorted-Array_v1.ts
function merge(nums1: number[], m: number, nums2: number[], n: number): void {
  const aux: number[] = [...nums1.slice(0, m), ...nums2];
  let i = 0;
  let j = m;
  for (let k = 0; k < nums1.length; k++) {
    if (i >= m) {
      nums1[k] = aux[j++];
    } else if (j >= aux.length) {
      nums1[k] = aux[i++];
    } else if (aux[i] < aux[j]) {
      nums1[k] = aux[i++];
    } else {
      nums1[k] = aux[j++];
    }
  }
}

File: Two_Pointers/088.Merge-Sorted-Array/088.Merge-Sorted-Array_v1.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int[] aux = new int[nums1.length];

        for (int i = 0; i < m; i++) {
            aux[i] = nums1[i];
        }
        for (int i = 0; i < n; i++) {
            aux[m + i] = nums2[i];
        }

        int i = 0;
        int j = m;
        
        for (int p = 0; p < m + n; p++) {
            if (i >= m) {
                nums1[p] = aux[j++];
            } else if (j >= m + n) {
                nums1[p] = aux[i++];
            } else if (aux[i] <= aux[j]) {
                nums1[p] = aux[i++];
            } else {
                nums1[p] = aux[j++];
            }
        }
    }
}